src/charm.py

```diff
@@ -532,15 +564,43 @@ def _on_set_password(self, event: ActionEvent) -> None:
                 event.fail(f"Failed changing the password: {e}")
                 return
 
-        self.set_secret("app", MongoDBUser.get_password_key_name_for_user(username), new_password)
+        secret_id = self.set_secret(
+            APP_SCOPE, MongoDBUser.get_password_key_name_for_user(username), new_password
+        )
 
         if username == BackupUser.get_username():
             self._connect_pbm_agent()
 
         if username == MonitorUser.get_username():
             self._connect_mongodb_exporter()
 
-        event.set_results({Config.Actions.PASSWORD_PARAM_NAME: new_password})
+        event.set_results(
+            {Config.Actions.PASSWORD_PARAM_NAME: new_password, "secret-id": secret_id}
+        )
+
+    def _on_secret_remove(self, event: SecretRemoveEvent):
+        # We are keeping this function empty on purpose until the issue with secrets
+        # is not fixed. The issue is: https://bugs.launchpad.net/juju/+bug/2023364
+        logging.error(
+            f"_on_secret_remove: Secret {event._id} seems to have no observers, could be removed"
```

Comment:
```suggestion
            f"_on_secret_remove: Secret {event._id} seems to have no observers, could not be removed"
```

Reply:
@MiaAltieri no, the message is correct as it is :-) 

The point is this. When the `secret-remove` event is raised, it's a "message" to the secret owner, that it could safely get rid of the secret completely. Currently we're ignoring this event (just logging that it happened). But the meaning of the event is that the secret is ready to be removed by the owner (whom at this point may decide what it wants to do).

NOTE that the behavior was broken not that long ago -- so safer not to take any action on this event so far I guess :-)